### PR TITLE
OTE SDK integration

### PR DESCRIPTION
**Pull request by sovrasov**

### CODE REVIEW

File: configs/ote_custom_classification/mobilenet_v3_small/aux_model.yaml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  root: './'
+  sources: ['data_train']
+  targets: ['data_val']
+  height: 224
+  width: 224
+  norm_mean: [0.485, 0.456, 0.406]
+  norm_std: [0.229, 0.224, 0.225]
+  save_dir: 'path/to/output_dir'
+  min_samples_per_id: 1
+  transforms:
```
Comment (kprokofi) [inline configs/ote_custom_classification/mobilenet_v3_small/aux_model.yaml]: This config looks very old. Could you use it as for mobilenetv3 large 1?

File: configs/ote_custom_classification/mobilenet_v3_small/main_model.yaml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  weight_decay: 5e-4
+  batch_size: 128
+  fixbase_epoch: 0
+  lr_scheduler: 'reduce_on_plateau'
+  early_stoping: True
+  train_patience: 5
+  patience: 4
+  gamma: 0.1
+
+sam:
```
Comment (kprokofi) [inline configs/ote_custom_classification/mobilenet_v3_small/main_model.yaml]: The same is for main_model. It is not runnable

File: configs/ote_custom_classification/mobilenetv2_imagenet.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
+model:
+  name: 'ptcv_mobilenetv2_w1'
```
Comment (kprokofi) [inline configs/ote_custom_classification/mobilenetv2_imagenet.yaml]: why do we need this model?

File: torchreid/data/datasets/image/classification.py [hunk 4]
```diff
@@ -117,7 +119,14 @@ def get_input(self, idx: int):
     def __getitem__(self, idx: int):
         input_image = self.get_input(idx)
         label = self.data_provider[idx]['label']
-        return input_image, label, 0
+        if isinstance(label, (tuple, list)): # when multi-label classification is available
```
Comment (kprokofi) [inline torchreid/data/datasets/image/classification.py:122]: Why have you moved this piece of code from the `ImageDataset` class? As I see, this class is inherited from `ImageDataset`

File: torchreid/engine/engine.py [hunk 5]
```diff
@@ -420,7 +420,7 @@ def run(
         if self.writer is not None:
             self.writer.close()
 
-        return top1
+        return self.best_metric
```
Comment (kprokofi) [inline torchreid/engine/engine.py]: For LR Finder we need to return the current metric, not the best one. Or do you propose to change the behavior and look at the best metric always?

File: torchreid/integration/sc/configuration.yaml [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+learning_parameters:
+  batch_size:
+    affects_outcome_of: TRAINING
+    default_value: 32
+    description: The number of training samples seen in each iteration of training.
+      Increasing this value improves training time and may make the training more
+      stable. A larger batch size has higher memory requirements.
+    editable: true
+    header: Batch size
+    max_value: 512
```
Comment (kprokofi) [inline torchreid/integration/sc/configuration.yaml:12]: For my interest, why is it 512?

File: torchreid/integration/sc/configuration.yaml [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+      action: DISABLE_EDITING
+      operator: AND
+      rules: []
+      type: UI_RULES
+    visible_in_ui: true
+    warning: null
+  max_num_epochs:
+    affects_outcome_of: TRAINING
+    default_value: 200
+    description: Increasing this value causes the results to be more robust but training
```
Comment (kprokofi) [inline torchreid/integration/sc/configuration.yaml:45]: It doesn't matter when the early stopping is on, though

File: torchreid/models/efficient_net_pytcv.py [hunk 8]
```diff
@@ -371,20 +371,22 @@ def forward(self, x, return_featuremaps=False, get_embeddings=False):
...
 
             y = self.features(x)
-            if return_featuremaps:
+            if return_featuremaps and not self.is_classification():
                 return y
 
             glob_features = self._glob_feature_vector(y, self.pooling_type, reduce_dims=False)
             logits = self.output(glob_features.view(x.shape[0], -1))
 
+            if return_featuremaps and self.is_classification():
```
Comment (kprokofi) [inline torchreid/models/efficient_net_pytcv.py]: This is not compatible with Q2L and the new GCN approach. I need to infer the model and get the features only.  But, now excess operations are done. We need to think about how to get this around.

**Review by kprokofi — COMMENTED** [review]
First comments

File: torchreid/engine/engine.py (hunk 5 above)
Comment (sovrasov) [reply torchreid/engine/engine.py]: Restored old behavior by returning the latest top1 as well

File: torchreid/models/efficient_net_pytcv.py (hunk 8 above)
Comment (sovrasov) [reply torchreid/models/efficient_net_pytcv.py]: Additional parameter to dump intermediate tensors was added

File: configs/ote_custom_classification/mobilenet_v3_large_075/main_model_multilabel.yaml [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+    cutout:
+      enable: True
+      cutout_factor: 0.35
+      p: 0.35
+    augmix:
+      enable: True
+      cfg_str: "augmix-m4-w2"
+
+loss:
+  name: 'softmax'
```
Comment (kprokofi) [inline configs/ote_custom_classification/mobilenet_v3_large_075/main_model_multilabel.yaml]: This config isn't for multilabel classification

File: configs/ote_custom_classification/mobilenet_v3_large_1/template.yaml [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+hyper_parameters:
+  base_path: "../../../torchreid/integration/sc/configuration.yaml"
+  parameter_overrides:
+    learning_parameters:
+      batch_size:
+        default_value: 30
+      max_num_epochs:
+        default_value: 200
+      learning_rate:
+        default_value: 0.001
```
Comment (kprokofi) [inline configs/ote_custom_classification/mobilenet_v3_large_1/template.yaml:36]: This default lr will destroy multilabel classification, unfortunately. But I will try to change the  ASL loss value to fit bigger learning rate values.

Comment (sovrasov) [reply configs/ote_custom_classification/mobilenet_v3_large_1/template.yaml:36]: Perhaps multiplying ASL by 0.1 could help to align it's magnitude with SM.

**Review by kprokofi — APPROVED** [review]
